### ISSUE TITLE
Add `compute` method to easily run a method in an isolate

### ISSUE THREAD

**Issue by nioncode** [dart-lang/sdk#40238]
This is basically a re-post of https://github.com/flutter/flutter/issues/48673, since I was advised to file this on this project instead.

It would be great if dart would offer a standard library method for simply running a method in an isolate, i.e. something along the lines of:
```dart
Future<R> run_background<R, P> (FutureOr<R> function(P argument), P argument) async {
  IsolateRunner iso = await IsolateRunner.spawn();
  try {
    return await iso.run(function, argument);
  } finally {
    await iso.close();
  }
}
```

Please note that this example uses [IsolateRunner.run](https://pub.dev/documentation/isolate/latest/isolate.isolate_runner/IsolateRunner/run.html) from the [isolate](https://pub.dev/packages/isolate) package.

Flutter provides the [compute()](https://api.flutter.dev/flutter/foundation/compute.html) method for this.
However, in order to create a library that can also be run in a pure dart setting (i.e. without flutter), we need a method that behaves like `compute` in the dart sdk.

Is the method I showed above good enough or are there any disadvantages when using `IsolateRunner.run` instead of `compute`? For a standard library implementation, we should probably get rid of `IsolateRunner` though or merge it into the sdk.

One nice thing about `IsolateRunner` is that we can keep a long lived instance of it and dispatch methods to the isolate when needed and not incur the cost of spawning up an isolate.

**Comment by lrhn** [comment]
The `dart:isolate` library is a fairly low-level library with very few convenience methods, which is why the `isolate` package exists with a more directly usable API.
I'd just use the `isolate` package, that's what it's there for. I'm sure it's possible to write code that is slightly smaller and faster than the more general `IsolateRunner` code ... but not much faster because most of the overhead comes from creating the isolate.

**Comment by MisterJimson** [comment]
The [isolate](https://pub.dev/packages/isolate) package is now discontinued. Curious how that impacts your guidance above @lrhn.

**Comment by lrhn** [comment]
I'm seriously considering adding a static `run` method to `isolate`, like
```dart
 static Future<T> run<T>(FutureOr<T> Function() computation);
```
which runs a function in a new (`Isolate.spawn`'ed) isolate.

With the improvements the VM has made on passing closures and other tricky values to other isolates, and with the `Isolate.exit` function to send the result back, it seems like the time is ripe for an efficient implementation.

**Comment by lrhn** [comment]
The `Isolate.run` method is released in Dart 2.19.

**Comment by mkustermann** [comment]
We have now a `Isolate.run()` method. So I think we can close this issue.
